pennylane/interfaces/batch/__init__.py

```diff
@@ -116,8 +251,13 @@ def cost_fn(params, x):
             gradient_fn = None
 
         elif mode == "backward":
+            # disable caching on the forward pass
+            execute_fn = cache_execute(device.batch_execute, cache=None)
+
             # replace the backward gradient computation
-            gradient_fn = device.gradients
+            gradient_fn = cache_execute(
+                device.gradients, cache, pass_kwargs=True, return_tuple=False
+            )

```

Comment:
Probably my unfamiliarity with the recent changes, but do we expect to need caching for device-based gradients? I thought this was mainly for parameter shift.

Reply:
Caching is only needed for device-based gradients if `mode="backwards"`. Backwards mode essentially means:

- On the forward pass, only the cost function is computed
- The gradients are only requested during backpropagation

This means that there will always be 1 additional eval required -- caching therefore reduces the number of evals by 1 😆 

Worth it?

I mean, I'd expect 99% of users to use device gradients with `mode="forward"`.

Reply:
Sounds good!

Reply:
Does this supersede https://github.com/PennyLaneAI/pennylane/pull/1341?

Reply:
No this complements it for now 🙂 

#1341 added the `use_device_state` keyword argument which instructs `QubitDevice.adjoint_jacobian()` to use the existing device state and avoid a redundant forward pass.

When `mode="forward"`, we can pass this option:

```python
execute(
    tapes,
    dev,
    gradient_fn="device",
    interface="torch",
    gradient_kwargs={"method": "adjoint_jacobian", "use_device_state": True},
    mode="forward"
)
```